src/main/java/org/datadog/jenkins/plugins/datadog/model/DatadogLinkAction.java

```diff
@@ -0,0 +1,70 @@
+package org.datadog.jenkins.plugins.datadog.model;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import hudson.model.Action;
+import org.datadog.jenkins.plugins.datadog.util.conversion.DatadogConverter;
+import org.datadog.jenkins.plugins.datadog.util.conversion.VersionedConverter;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import static org.datadog.jenkins.plugins.datadog.util.conversion.VersionedConverter.ignoreOldData;
+
+public class DatadogLinkAction implements Action {
+
+    private final String url;
+
+    public DatadogLinkAction(BuildData buildData, String siteName) {
+        String query = String.format("ci_level:pipeline @ci.pipeline.name:\"%s\" @ci.pipeline.number:%s", buildData.getJobName(), buildData.getBuildNumber(""));
+        String urlEncodedQuery = URLEncoder.encode(query, StandardCharsets.UTF_8);
+        this.url = String.format("https://app.%s/ci/pipeline-executions?query=%s", siteName, urlEncodedQuery);
```

Comment:
Notice that some orgs can have dedicated urls that are not the default `app.%s`. 
Using this link won't work for them. Should we add this limitation somewhere?

Reply:
Ahh, that is a good catch! I don't think there's a way to work around this, knowing only the site.
I asked around and did some tests of my own: the custom links indeed seem to replace the standard ones, not just be an alias in addition to them.

So what I ended up doing is adding a dedicated "Datadog App hostname" field to the plugin configuration.
The logic is as follows:
- if the field is configured, we use its value to construct the link URL
- if the field is empty, we try to infer the hostname using the intake URLs - this option will only work for agentless config
- if agentful connectivity is configured (and the Datadog App hostname is not provided), we will not show the link at all